### PR TITLE
First pass removal of Backbone and lodash.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

*Short description*

## TODO

- [X] Have tests been written for the new code?
- [X] New dependencies (if any) added to requirements file
- [X] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

Removed Backbone and Lodash from all user facing code. Lodash is possibly still used in the build process.

Also changed dependencies as appropriate (noticed that log-level was in the devDependencies, but it is built into frontend code).

### CODE REVIEW

File: karma_config/karma.conf.js [hunk 1]
```diff
@@ -72,7 +72,8 @@ module.exports = function(config) {
     eslint: {
       engine: {
         configFile: '.eslintrc.js'
-      }
+      },
+      stopOnError: false
```
Comment (rtibbles) [inline karma_config/karma.conf.js:76]: Also added this, because it was annoying when the tests would die for linting.

**Comment by rtibbles** [comment]
This should be ready to merge now. The only outstanding thing for ES6 conversion is that eslint is throwing errors about camelCase - personally, I prefer snake case, as evidenced by the Mediator code, but can alter if desired.

**Comment by MCGallaspy** [comment]
Can we keep lodash? It's generally useful beyond being a dependency for Backbone.

**Comment by rtibbles** [comment]
It's quite bulky, unless we use the restricted subset that has reduced functionality. In addition, if we are using ES6 then some of those idioms should replace use of lodash.

It is still in the devDependencies (for the non ES6 build code), but if we want to use ES6, we should try to use ES6 instead of falling back to lodash methods.

**Comment by rtibbles** [comment]
Also, if you look at the diffs here, the removal of the lodash methods in favour of more ES6-y ones does not result in an explosion of code.

**Comment by MCGallaspy** [comment]
The full build is ~22kb and the "core" build is ~4kb. IMO it's useful enough even though there is overlap in ES6 to justify the full build. Regardless the particular case I was thinking of was comparing non-primitive types by value. Non-primitive types are compared by reference in javascript, so that for example:
```js
foo = [1, 2, 3];
bar = [1, 2, 3];
foo == bar; // false
foo === bar; // false again
```

The same holds for objects, Maps, Sets, etc. Lodash has an [isEqual](https://lodash.com/docs#isEqual) method in the core for deep comparison. There's other stuff in the full build that doesn't have a built-in ES6 analog AFAIK (like lots of stuff under the Function heading). So I think including either the core build or the full build would be a good "investment". My 2 cents.

**Comment by rtibbles** [comment]
I would rather have a concrete use case for these before putting extra libraries in the core (especially something that might have performance implications like a deepEquals).

Also, if there are particular methods, we can pick only those methods to include from lodash, rather than just including the entire library.

**Comment by MCGallaspy** [comment]
I do have a concrete use case. I use deep comparison in tests. But I misunderstood, I thought you were removing it from the dependencies. I see that you're just removing uses of it. That's fine IMO, up to the discretion of the developer.

**Comment by indirectlylit** [comment]
sounds good. Yeah there are things in lodash I can imagine using too (e.g. `_.throttle` and `_.debounce`) but would be more inclined to cherry pick those into the client code

**Pull request merged**